p2p/src/authenticated/config.rs

```diff
@@ -42,8 +42,6 @@ pub struct Config<C: Scheme> {
     pub allow_private_ips: bool,
 
     /// Maximum size allowed for messages over any connection.
-    ///
-    /// Messages larger than this size will be chunked.
     pub max_message_size: usize,
```

Comment:
We should make a note of this behavior in the comment (so user's are aware this is happening behind the scenes/in case they apply some other message pre-processing or something). I guess the alternative would be to find a way to avoid using Protobuf for these sort of things and encode `type` into `Stream` (but I think protobuf here is cleaner).

Reply:
This note should also indicate we also incorporate some overhead for ENCRYPTION_TAG. (so `message_size + PROTOBUF_OVERHEAD + ENCRYPTIONT_TAG`).